.github/workflows/ci.yaml

```diff
@@ -119,21 +155,73 @@ jobs:
           echo 'pantos-service-node-full pantos/service-node/signer_key_password string 1234' | sudo debconf-set-selections;
           echo 'pantos-service-node-full pantos/common/configure_postgresql boolean true' | sudo debconf-set-selections;
           echo 'pantos-service-node-full pantos/common/configure_rabbitmq boolean true' | sudo debconf-set-selections;
-          echo 'pantos-service-node-full pantos/common/reset_users boolean true' | sudo debconf-set-selections;
-          echo 'pantos-service-node-full pantos/common/start_services boolean false' | sudo debconf-set-selections;
+          echo 'pantos-service-node-full pantos/common/start_services boolean true' | sudo debconf-set-selections;
+
+      - name: Copy files from eth-data and bnb-data
+        run: |
+          sudo mkdir -p /etc/pantos
+          sudo chmod 777 /etc/pantos
+          docker compose cp eth:/data/keystore /etc/pantos/keystore.eth
+          docker compose cp eth:/data/ETHEREUM.env /etc/pantos/ETHEREUM.env
+          docker compose cp bnb:/data/keystore /etc/pantos/keystore.bnb
+          docker compose cp bnb:/data/BNB.env /etc/pantos/BNB.env
+        working-directory: ethereum-contracts
+
+      - name: Set env
+        run: |
+          sudo cp ./service-node-config.local.env /etc/pantos/service-node-config.env
+          # Append the new addresses from the copied env files
+          sudo cat /etc/pantos/ETHEREUM.env | sudo tee -a /etc/pantos/service-node-config.env > /dev/null
+          sudo cat /etc/pantos/BNB.env | sudo tee -a /etc/pantos/service-node-config.env > /dev/null
+          sudo chmod 644 /etc/pantos/service-node-config.env
 
       - name: Install Debian package
         run: |
           sudo apt-get update
-          sudo apt-get install -y ./build/*.deb
+          sudo apt-get install -y ./build/*.deb curl
 
       # Check service is running
       - name: Check service is running
+        run: |
+          # Wait for the service to start
+          while ! sudo systemctl is-active --quiet pantos-service-node-server; do
+            echo "Server is not running, retrying in 1 second."
+            sleep 1
+          done
+          echo "Server is running"
+          while ! sudo systemctl is-active --quiet pantos-service-node-celery; do
+            echo "Celery is not running, retrying in 1 second."
+            sleep 1
+          done
+          echo "Celery is running"
+          # Wait for curl to be positive
+          while true; do
```

Comment:
Is there any guarantee that we will get a 200 status code? Might it be, that we get some error code due to an error in the code and being stuck in the loop for 6 hours until the job gets killed?

Reply:
Fair point, thanks! I'll add a timeout